Common/StringExtensions.cs

```diff
@@ -0,0 +1,138 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Globalization;
+
+namespace QuantConnect
+{
+    /// <summary>
+    /// Provides extension methods for properly parsing and serializing values while properly using
+    /// an IFormatProvider/CultureInfo when applicable
+    /// </summary>
+    public static class StringExtensions
```

Comment:
Moved `Parse*` functions to a new static class, `Parse` to enable syntax such as:
``` csharp
var myDecimal = Parse.Decimal(myString);
```
and it will do the required thing. After looking at the usage of extensions for parsing, I felt it unnecessarily polluted string's method-space (intellisense), and also from a style perspective, using extension methods for parsing just seemed incorrect.